In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline

In [13]:
#sample data from Neuron. data = metadata, authors = author data, mesh = meshdata
data = pd.read_csv('lukeholman_gendergap/data/data/Neuron-data.csv', encoding="ISO-8859-1", error_bad_lines = False)
authors = pd.read_csv('lukeholman_gendergap/data/authors/Neuron-authors.csv', encoding="ISO-8859-1", error_bad_lines = False)
mesh = pd.read_csv('lukeholman_gendergap/data/mesh/Neuron-mesh.csv', encoding="ISO-8859-1", error_bad_lines = False)

In [190]:
print(data.head(10))
print(data.shape)

       pmid                                           abstract  year  \
0  31708306  Three-dimensional documentation of the axonal ...  2019   
1  31706697  The posterior parietal cortex (PPC) performs m...  2019   
2  31706696  The olfactory bulbs (OBs) are the first site o...  2019   
3  31704028  Fine orchestration of excitatory and inhibitor...  2019   
4  31697921  One of the most common brain tumors in childre...  2019   
5  31697920                                                NaN  2019   
6  31697919                                                NaN  2019   
7  31697918  How can we relate processing in the retina to ...  2019   
8  31697917  Perseverance in foraging is a high-risk/high-g...  2019   
9  31697916  In this issue of Neuron, Gharpure et al. (2019...  2019   

                                               title journal        country  
0  Holographic Reconstruction of Axonal Pathways ...  Neuron  United States  
1     Corticostriatal Flow of Action Selection Bias

In [191]:
print(authors.head(10))
print(authors.shape)

       pmid   ForeName  LastName Initials  order  \
0  31708306   Mikkel V  Petersen       MV    1.0   
1  31708306    Jeffrey    Mlakar        J    2.0   
2  31708306  Suzanne N     Haber       SN    3.0   
3  31708306     Martin    Parent        M    4.0   
4  31708306     Yoland     Smith        Y    5.0   
5  31708306    Peter L    Strick       PL    6.0   
6  31708306     Mark A  Griswold       MA    7.0   
7  31708306  Cameron C  McIntyre       CC    8.0   
8  31706697   Eun Jung     Hwang       EJ    1.0   
9  31706697   Trevor D      Link       TD    2.0   

                                         Affiliation  
0  Department of Biomedical Engineering, Case Wes...  
1  Interactive Commons, Case Western Reserve Univ...  
2  Department of Pharmacology and Physiology, Uni...  
3  Department of Psychiatry and Neuroscience, Uni...  
4  Department of Neurology, Emory University, Atl...  
5  Department of Neurobiology, University of Pitt...  
6  Interactive Commons, Case Western Reser

In [189]:
print(mesh.tail(10))
print(mesh.shape)

           pmid                     mesh
255400  2483091  Molecular Sequence Data
255401  2483091        Myelin P0 Protein
255402  2483091          Myelin Proteins
255403  2483091                 genetics
255404  2483091                     Rats
255405  2483091            Schwann Cells
255406  2483091               metabolism
255407  2483091            Sciatic Nerve
255408  2483091               metabolism
255409  2483091             Transfection
(255410, 2)


In [17]:
country = []
country = data['country'].value_counts()
print(country)

United States     10634
United Kingdom      496
Canada               53
International        27
Italy                23
Switzerland           3
Name: country, dtype: int64


In [18]:
forename = []
forename = authors['ForeName'].value_counts()
print(forename)

M                     687
A                     518
S                     482
J                     473
R                     370
K                     340
T                     334
H                     320
C                     314
P                     304
D                     278
G                     256
E                     243
David                 241
Michael               236
Y                     229
L                     215
B                     211
N                     181
Thomas                172
F                     165
Peter                 159
Daniel                130
Christian             118
I                     117
Martin                116
J A                    98
D J                    95
John                   93
W                      90
                     ... 
Arezu                   1
Kristofer W             1
Tingxu                  1
Vicky A                 1
Charles-Francois V      1
Chul Hoon               1
Stephen B H             1
Baubak      

In [24]:
#call genderize API
from genderize.genderize import Genderize

In [479]:
#clean the ForeName data: take out single character first names, first names plus single characters
import re

    
def getfirstname(word):    
    matchstr = r'\w{2,10000}'
    if isinstance(word, str) is False:
        return 'None'
    else:
        out = re.search(matchstr,word)
        if out is None:
            return 'None'
        return out.group()

names = [getfirstname(word) for word in authors['ForeName']]

# for word in authors['ForeName']:
#     if getfirstname(word) is None:
#         print(word)
    


In [480]:
names
names_count = pd.DataFrame(data = names)
names_count
name_count = []
name_count = names_count[0].value_counts()
print(name_count)

None           13510
Michael          744
David            743
John             483
Thomas           476
Daniel           343
Robert           338
Richard          317
Mark             313
Peter            307
James            280
Matthew          264
Christopher      263
William          229
Andrew           223
Paul             221
Eric             205
Stephen          192
Steven           186
Martin           169
Joseph           164
Jeffrey          161
Christian        160
Jonathan         152
Timothy          149
Joshua           146
Kevin            143
Alexander        139
Benjamin         138
Brian            133
               ...  
Leann              1
Gerit              1
Ruiqi              1
Evangelos          1
Poorva             1
Chaoran            1
Jacobo             1
Zoë                1
Danko              1
Crissy             1
Pal                1
Sungkyung          1
Jigar              1
Unikora            1
Jaeseob            1
Floris             1
Kyunghee     

In [491]:
#make a list of names to work with. either sequentially or randomly
#call genderize API without API Key

#get names
names_test = []
names_test = names[49900:50000]
#names_test = np.random.choice(names, 100)
print(names_test)   
np.array(names_test)
print(names_test)
    
#gender_names_test = []
#gender_names_test = Genderize().get([names_test])
#print(Genderize().get([names_test]))

['None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None']
['None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None'

In [301]:
df = pd.DataFrame()
print(df)

Empty DataFrame
Columns: []
Index: []


In [469]:
#make chunks of 10 and iterate through
#establish interval for probability of threshold 
import math 

numcalls = math.ceil(len(names_test)/10)
print(numcalls)
gender_names_test = [] 
names_batch = []
genders = []

genders_name = []
genders_gender = []
genders_prob = []

# for i in range(0,numcalls):
#     names_test2 = names_test[i*10 : i*10+10]
#     print(names_test2)
#     gender_names_test = Genderize().get([names_test2])
#     print(gender_names_test)
#     for value in range(0,len(gender_names_test)):
#         if (gender_names_test[value]['probability']) > 0.85:
#             genders.append((gender_names_test[value]['gender']))
#         else:
#             genders.append(np.nan)
    
    
#     #names_batch.append(gender_names_test)
# #print(names_batch)
# print(genders)


for i in range(0,numcalls):
    names_test2 = names_test[i*10 : i*10+10]
    print(names_test2)
    gender_names_test = Genderize().get([names_test2])
    #print(gender_names_test)
    for value in range(0,len(gender_names_test)):
            genders_name.append((gender_names_test[value]['name']))
            genders_gender.append((gender_names_test[value]['gender']))
            genders_prob.append((gender_names_test[value]['probability']))
    
    
#names_batch.append(gender_names_test)
#print(names_batch)
#print(genders)
print(genders_name)
print(genders_gender)
print(genders_prob)
    

10
[None, None, None, None, None, None, None, None, None, None]


GenderizeException: ("Missing 'name' parameter", 422, {'Server': 'nginx/1.16.1', 'Date': 'Mon, 25 Nov 2019 03:44:01 GMT', 'Content-Type': 'application/json; charset=utf-8', 'Content-Length': '36', 'Connection': 'keep-alive', 'Access-Control-Allow-Origin': '*', 'Access-Control-Allow-Methods': 'GET', 'Access-Control-Allow-Headers': 'Content-Type, X-Genderize-Source', 'ETag': 'W/"24-8HOPqNcFFlm/vxurwlbjOfqPxvc"'})

In [492]:
#make chunks of 10 and iterate through
import math 

genders = []
numcalls = math.ceil(len(names_test)/10)
print(numcalls)

# genderize = Genderize(
#     #user_agent='GenderizeDocs/0.0',
#     api_key='3facc1b5e8f93db3ac7d313d56f591d1',
#     timeout=5.0)
#print(genderize.get([names_test]))

# for i in range(0,numcalls):
#     names_test2 = names_test[i*10 : i*10+10]
#     #print(names_test2)
#     gender_names_test = Genderize().get([names_test2])
#     print(gender_names_test)
#     for value in range(0,len(gender_names_test)):
#         if (gender_names_test[value]['probability']) > 0.93:
#             genders.append((gender_names_test[value]['gender']))
#         else:
#             genders.append('uncertain')

for i in range(0,numcalls):
    names_test2 = names_test[i*10 : i*10+10]
    print(names_test2)
    for name in range(0, len(names_test2)):
        if names_test[name] != 'None':
            gender_names_test = Genderize().get([names_test2[name]])
            print(gender_names_test)
            for value in range(0,len(gender_names_test)):
                if (gender_names_test[value]['probability']) > 0.93:
                    genders.append((gender_names_test[value]['gender']))
                else:
                    genders.append('uncertain')
        else:
            genders.append('uncertain')
    
    
    #names_batch.append(gender_names_test)
#print(names_batch)
print(genders)

    

10
['None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None']
['None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None']
['None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None']
['None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None']
['None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None']
['None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None']
['None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None']
['None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None']
['None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None']
['None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None', 'None']
['uncertain', 'uncertain', 'uncertain', 'uncertain', 'uncertain', 'uncertain', 'uncertain', 'uncertain', 'uncertain', 'uncertain', 'uncertain', 'uncertain', 'uncertain', 'uncertain', 'unc

In [490]:
names_test[8]

'None'

In [315]:
#setting up initial df to compare handscore and genderize prob, then append to initial df
df['name'] = genders_name
df['gender'] = genders_gender
df['probability'] = genders_prob


genders_handscore = ['male', 'male', 'male', 'unknown', 'female', 'unknown', 'male', 'male', 'unknown', 'male',
                     'female', 'male', 'male', 'male', 'female', 'female', 'male', 'male', 'female', 'female',
                     'unknown', 'male', 'male', 'male', 'male', 'female', 'male', 'male', 'male', 'male',
                     'unknown', 'male', 'unknown', 'male', 'unknown', 'male', 'male', 'female', 'unknown', 'male',
                     'male', 'unknown', 'female', 'female', 'unknown', 'male', 'female', 'female', 'male', 'male',
                     'male', 'male', 'female', 'male', 'male', 'unknown', 'male', 'female', 'male', 'male',
                     'unknown', 'unknown', 'male', 'female', 'female', 'male', 'female', 'female', 'male', 'female',
                     'male', 'male', 'male', 'male', 'male', 'male', 'male']


df['handscore'] = genders_handscore
with pd.option_context('display.max_rows', None, 'display.max_columns', None): 
    display(df)

name  gender  probability handscore
0      Mikkel    male         0.99      male
1     Jeffrey    male         0.99      male
2     Suzanne  female         0.98    female
3      Martin    male         0.98      male
4      Yoland    male         0.54   unknown
5       Peter    male         0.99      male
6        Mark    male         0.99      male
7     Cameron    male         0.92      male
8         Eun  female         0.82    female
9      Trevor    male         0.99      male
10     Yvonne  female         0.98    female
11       Shan    male         0.68   unknown
12       Eric    male         0.99      male
13     Varoth    None         0.00      male
14       Sage    male         0.73    female
15     Keelin  female         0.53   unknown
16      Byung    male         0.99   unknown
17     Takaki    male         0.97      male
18       Tali  female         0.60    female
19      Timna  female         1.00    female
20       Lior    male         0.86      male
21      Sagit    male         0.68      male
22       Kobi    male         0.94      male
23       Reut  female         0.77    female
24       Edna  female         0.98    female
25      Thijs    male         0.99      male
26       Noam    male         0.94      male
27     Matteo    male         0.99      male
28       Nora  female         0.97    female
29    Olivier    male         0.99      male
30    Sabrina  female         0.98    female
31    Florent    male         0.99      male
32  Guillaume    male         0.99      male
33    Damarys  female         0.96    female
34     Cécile    None         0.00    female
35      David    male         0.99      male
36     Helmut    male         0.99    female
37     Jerome    male         0.99      male
38      David    male         0.99      male
39      Boris    male         0.99      male
40   Yongrong  female         0.67   unknown
41     Thomas    male         0.99      male
42     Thomas    male         0.99      male
43     Pierre    male         0.99      male
44      Henry    male         0.98      male
45     Dennis    male         0.99      male
46     Joanna  female         0.98    female
47     Casper    male         0.95      male
48    Claudia  female         0.98    female
49       Rein    male         0.84      male
50      Joris    male         0.98      male
51        Jan    male         0.95      male
52     Pascal    male         0.99      male
53       Ruud    male         0.99      male
54   Matthijs    male         0.99      male
55     Justin    male         0.98      male
56      Wenbo    male         0.85      male
57   Shantanu    male         1.00   unknown
58    Taihong    male         1.00      male
59    Fengyun    male         0.55   unknown
60    Wenxing    male         0.83   unknown
61         He    male         0.68   unknown
62    Antonio    male         0.99      male
63      Diana  female         0.98    female
64      Abdul    male         0.98      male
65        Joy  female         0.84    female
66    QueeLim    None         0.00   unknown
67    Rebecca  female         0.98    female
68        Yun  female         0.52    female
69      Ignas    male         0.99      male
70     Jochen    male         1.00   unknown
71   Johannes    male         0.99      male
72      David    male         0.99      male
73    Hongbin    male         0.98   unknown
74     Fawwad    male         0.97      male
75      James    male         0.99      male
76     Sophie  female         0.98    female
77     Varoth    None         0.00      male
78      Byung    male         0.99   unknown
79      Csaba    male         0.99      male
80    Stephan    male         0.99      male
81      James    male         0.99      male
82      Gregg    male         0.99      male
83       Alex    male         0.90      male
84    Aneesha  female         0.93    female
85  Alexander    male         0.99      male
86   Nicholas    male         0.99      male
87     Sliman    male         0.96      male
88  

In [421]:
# df.append(df1) 
# with pd.option_context('display.max_rows', None, 'display.max_columns', None): 
#     display(df)
# len(df['name'])

frames = [df, df1, df2, df3, df4, df5, df6, df7, df8, df9]
result = pd.concat(frames, ignore_index=True)
display(result)

len(result['name'])

name  gender  probability handscore
0         Mikkel    male         0.99      male
1        Jeffrey    male         0.99      male
2        Suzanne  female         0.98    female
3         Martin    male         0.98      male
4         Yoland    male         0.54   unknown
5          Peter    male         0.99      male
6           Mark    male         0.99      male
7        Cameron    male         0.92      male
8            Eun  female         0.82    female
9         Trevor    male         0.99      male
10        Yvonne  female         0.98    female
11          Shan    male         0.68   unknown
12          Eric    male         0.99      male
13        Varoth    None         0.00      male
14          Sage    male         0.73    female
15        Keelin  female         0.53   unknown
16         Byung    male         0.99   unknown
17        Takaki    male         0.97      male
18          Tali  female         0.60    female
19         Timna  female         1.00    female
20          Lior    male         0.86      male
21         Sagit    male         0.68      male
22          Kobi    male         0.94      male
23          Reut  female         0.77    female
24          Edna  female         0.98    female
25         Thijs    male         0.99      male
26          Noam    male         0.94      male
27        Matteo    male         0.99      male
28          Nora  female         0.97    female
29       Olivier    male         0.99      male
..           ...     ...          ...       ...
765  Christopher    male         0.99      male
766          Wen  female         0.54   unknown
767        Janet  female         0.98      male
768       Istvan    male         0.99   unknown
769      Michael    male         0.99      male
770        Kosei    male         0.96      male
771       Edward    male         0.99      male
772  Christopher    male         0.99      male
773         Ying  female         0.86   unknown
774        Qihao    male         0.67   unknown
775     Philippe    male         0.99      male
776      Nicolas    male         0.99      male
777     Masahiro    male         1.00      male
778       Guojun    male         1.00   unknown
779         Miri  female         0.70    female
780     Ravindra    male         0.99   unknown
781    Franziska  female         0.97    female
782       Ernest    male         0.98      male
783         Rona  female         0.92    female
784        Weiyu    male         0.67   unknown
785       Thomas    male         0.99      male
786        Nihar    male         0.94      male
787        Kevin    male         0.99      male
788      Akihiro    male         1.00      male
789        Diego    male         0.99      male
790       Ahryon    None         0.00      male
791     Geoffrey    male         0.99      male
792          Gil    male         0.92      male
793      Kathryn  female         0.98    female
794         Lily  female         0.98    female

[795 rows x 4 columns]

795

In [422]:
prob_wrong = []

for i in range(0, len(df9['name'])):
    if df9['gender'][i] != df9['handscore'][i]:
        #print(df['probability'][i])
        prob_wrong.append(df9['probability'][i])
    else: 
        None 

#take out zeros: means there is nothing (will bring down average)
prob_wrong = [i for i in prob_wrong if i != 0.0]

print(prob_wrong)
print(sum(prob_wrong)/len(prob_wrong))
print(np.std(prob_wrong))
print(len(prob_wrong))

#df =  avg 0.78 std 0.17 n 22
#df1 = avg 0.86 std 0.15 n 17
#df2 = avg 0.86 std 0.15 n 5
#df3 = avg 0.84 std 0.13 n 11
#df4 = avg 0.79 std 0.15 n 18
#df5 = avg 0.84 std 0.13 n 7
#df6 = avg 0.80 std 0.14 n 13
#df7 = avg 0.82 std 0.15 n 14
#df8 = avg 0.77 std 0.21 n 8
#df9 = avg 0.85 std 0.15 n 16


[0.99, 0.65, 0.8, 0.96, 0.73, 0.9, 1.0, 0.99, 0.54, 0.98, 0.99, 0.86, 0.67, 1.0, 0.99, 0.67]
0.8575
0.15245901088489325
16


In [379]:
import scipy.stats


def mean_confidence_interval(data, confidence=0.95):
    a = 1.0 * np.array(data)
    n = len(a)
    m, se = np.mean(a), scipy.stats.sem(a)
    h = se * scipy.stats.t.ppf((1 + confidence) / 2., n-1)
    return m, m-h, m+h

In [423]:
mean_confidence_interval(prob_wrong, confidence = 0.95)

(0.8574999999999999, 0.7735960271659204, 0.9414039728340795)

#CI 0.95
df (0.7777272727272728, 0.7007993023978496, 0.854655243056696)
df1 (0.861764705882353, 0.7808727372688707, 0.9426566744958352)
df2 (0.858, 0.6499703227226927, 1.0660296772773072)
df3 (0.8354545454545454, 0.7416721243222468, 0.9292369665868441)
df4 (0.7938888888888889, 0.7147418262376936, 0.8730359515400842)
df5 (0.8428571428571429, 0.7101474942527943, 0.9755667914614914)
df6 (0.7984615384615386, 0.7126870495325227, 0.8842360273905544)
df7 (0.8242857142857145, 0.7338482006528023, 0.9147232279186267)
df8 (0.7662500000000001, 0.5758732271494678, 0.9566267728505324)
df9 (0.8574999999999999, 0.7735960271659204, 0.9414039728340795)

In [439]:
#(0.85 + 0.94 + 1.06 + 0.93 + 0.87 + 0.98 + 0.88 + 0.91 + 0.96 + 0.94)/10
(0.78 + 0.86 + .86 + 0.83 + 0.79 + 0.84 + 0.80 + 0.82 + 0.77 + 0.86)/10

0.821

In [ ]:
#make chunks of 10 and iterate through
import math 

numcalls = math.ceil(len(names_test)/10)
print(numcalls)
gender_names_test = [] 
names_batch = []
genders = []

genders_name = []
genders_gender = []
genders_prob = []

# for i in range(0,numcalls):
#     names_test2 = names_test[i*10 : i*10+10]
#     print(names_test2)
#     gender_names_test = Genderize().get([names_test2])
#     print(gender_names_test)
#     for value in range(0,len(gender_names_test)):
#         if (gender_names_test[value]['probability']) > 0.85:
#             genders.append((gender_names_test[value]['gender']))
#         else:
#             genders.append(np.nan)
    
    
#     #names_batch.append(gender_names_test)
# #print(names_batch)
# print(genders)


In [236]:
#call genderize API with API key, gives 1,000,000 names in a month
#PROBLEM: only calls 10 at a time, so need to iterate 10 at a time
#with no API: 1000 calls a day
genderize = Genderize(
    #user_agent='GenderizeDocs/0.0',
    #api_key='3facc1b5e8f93db3ac7d313d56f591d1',
    timeout=5.0)
print(genderize.get([names_test]))

GenderizeException: ("Missing 'name' parameter", 422, {'Server': 'nginx/1.16.1', 'Date': 'Sun, 24 Nov 2019 01:37:46 GMT', 'Content-Type': 'application/json; charset=utf-8', 'Content-Length': '36', 'Connection': 'keep-alive', 'Access-Control-Allow-Origin': '*', 'Access-Control-Allow-Methods': 'GET', 'Access-Control-Allow-Headers': 'Content-Type, X-Genderize-Source', 'ETag': 'W/"24-8HOPqNcFFlm/vxurwlbjOfqPxvc"'})

In [460]:
authors_test = authors[49900:500310]
print(authors_test)

          pmid ForeName        LastName Initials  order  \
49900  9491984        S  de la Escalera        S    5.0   
49901  9491984        F         Jiménez        F    6.0   
49902  9491983      H H              Yu       HH    1.0   
49903  9491983      H H            Araj       HH    2.0   
49904  9491983      S A           Ralls       SA    3.0   
49905  9491983      A L        Kolodkin       AL    4.0   
49906  9491982      E F          Espejo       EF    1.0   
49907  9491982      R J         Montoro       RJ    2.0   
49908  9491982      J A        Armengol       JA    3.0   
49909  9491982        J    López-Barneo        J    4.0   
49910  9491981      D L        Schacter       DL    1.0   
49911  9491981      R L         Buckner       RL    2.0   
49912  9491980      W N         Frankel       WN    1.0   
49913  9491979      R R          Fuller       RR    1.0   
49914  9491979      L L           Moroz       LL    2.0   
49915  9491979        R        Gillette        R    3.0 

In [428]:
#if probability is greater than 0.85, use it...but need to randomly sample and check if this is a good probability
genders = []
for value in range(0,len(gender_names_test)):
    if (gender_names_test[value]['probability']) > 0.93:
        genders.append((gender_names_test[value]['gender']))
    else:
        genders.append(np.nan)

In [429]:
print(genders)

[nan, nan, nan, nan, 'male', nan, 'male', nan, 'male', 'male']


In [182]:
prob = []
for value in range(0,len(gender_names_test)):
    prob.append((gender_names_test[value]['probability']))

In [183]:
print(prob)

[0.99, 0.99, 0.98, 0.98, 0.54, 0.99, 0.99, 0.92, 0.82, 0.99]


In [184]:
prob[0] > 0

True

In [504]:
# make new column for gender
authors_test['gender'] = genders
print(authors_test)
print(authors_test.dtypes)
authors_test.to_csv('lukeholman_gendergap/data/gender_data/authors_test.csv')

          pmid ForeName        LastName Initials  order  \
49900  9491984        S  de la Escalera        S    5.0   
49901  9491984        F         Jiménez        F    6.0   
49902  9491983      H H              Yu       HH    1.0   
49903  9491983      H H            Araj       HH    2.0   
49904  9491983      S A           Ralls       SA    3.0   
49905  9491983      A L        Kolodkin       AL    4.0   
49906  9491982      E F          Espejo       EF    1.0   
49907  9491982      R J         Montoro       RJ    2.0   
49908  9491982      J A        Armengol       JA    3.0   
49909  9491982        J    López-Barneo        J    4.0   
49910  9491981      D L        Schacter       DL    1.0   
49911  9491981      R L         Buckner       RL    2.0   
49912  9491980      W N         Frankel       WN    1.0   
49913  9491979      R R          Fuller       RR    1.0   
49914  9491979      L L           Moroz       LL    2.0   
49915  9491979        R        Gillette        R    3.0 

/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  


In [474]:
len(genders)

131

In [509]:
import time

print('lukeholman_gendergap/data/gender_data/'+ time.strftime('%Y-%m-%d_%H-%S') + '.csv')

lukeholman_gendergap/data/gender_data/2019-11-25_20-07.csv
